M2/libraries/Makefile.in

```diff
@@ -22,26 +22,27 @@ $(foreach d,@LIBLIST@ @PROGLIST@,		\
 		$(eval .PHONY: $t-all)))
 
 uninstall:; rm -rf */.installed* $(BUILTLIBPATH)
-all:install-programs
+all:install-programs install-licenses
 install-programs:
 	: installing programs in @pre_programsdir@
-	if [ -d $(BUILTLIBPATH)/bin/. ] ; \
-	then rm -rf @pre_programsdir@ ; \
-	     @INSTALL@ -d @pre_programsdir@ ; \
-	     @TAR@ cf - -C $(BUILTLIBPATH)/bin/. . | @TAR@ xf - -C @pre_programsdir@ ; \
-	fi
+	$(MKDIR_P) $(BUILTLIBPATH)/bin ; \
+	@INSTALL@ -d @pre_programsdir@ ; \
+	@TAR@ cf - -C $(BUILTLIBPATH)/bin/. . | @TAR@ xf - -C @pre_programsdir@ ;
+install-licenses:
+	: installing library and submodule licenses in @pre_licensesdir@
+	$(MKDIR_P) $(BUILTLIBPATH)/licenses ; \
+	@INSTALL@ -d @pre_licensesdir@ ; \
+	@TAR@ cf - -C $(BUILTLIBPATH)/licenses/. . | @TAR@ xf - -C @pre_licensesdir@ ;
 install-dynamic-libraries:
 	: installing sharable library files in @pre_librariesdir@
-	if test -d $(BUILTLIBPATH)/lib ;				     \
-	then rm -rf @pre_librariesdir@ ;				     \
```

Comment:
Since you had problems with the command `rm -rf @pre_librariesdir@`, I suspect cmake is building the libraries directly into `usr-dist` instead of into `usr-host`.  That wouldn't be so good, because then we'd be distributing library include files with Macaulay2 binaries.  Is it so?

Reply:
It's copying the binaries directly in the programs directory when they are built, but only the programs, nothing else.

Reply:
Oh, so that's why you had to modify the makefiles, because you want to do part of their job.

A non-intrusive way to do it, without modifying the makefiles, would be to put the libraries somewhere else, and just add CPPFLAGS=-I/somewhere/else/include and LDFLAGS=-L/somewhere/else/lib to the configure command line.